scripts/manual_deploy/00_treasury_vault_logic.ts

```diff
@@ -0,0 +1,105 @@
+import { run } from "hardhat";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import {
+  CHAINID,
+  USDC_ADDRESS,
+  OTOKEN_FACTORY,
+  GAMMA_CONTROLLER,
+  MARGIN_POOL,
+  GNOSIS_EASY_AUCTION,
+  WETH_ADDRESS,
+  TD_OTOKEN_FACTORY,
+  TD_CONTROLLER,
+  TD_MARGIN_POOL,
+} from "../../constants/constants";
+
+interface args {
+  protocol: "10D" | "Gamma"; // Will deploy using 10D or Gamma OToken factory, controller and margin pool respectively
```

Comment:
For when we want to deploy using 10D or Gamma, and can also be extended in the future for optionality for different lifecycles if needed